### ISSUE TITLE
theme-live-codeblock throws and error when live code uses optional chaining

### ISSUE THREAD

**Issue by csantos-nydig** [facebook/docusaurus#10248]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

Users see `SyntaxError: Unexpected token, expected "," (1:109)` error when they use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) in their live code

### Reproducible demo

https://codesandbox.io/p/devbox/prod-monad-7gh39v?file=%2Fdocusaurus.config.ts

### Steps to reproduce

1. Start server (dev or build, doesn't matter)
2. Navigate to "Tutorial intro" where the live example lives
3. Notice the error

### Expected behavior

live example should render without any errors; similar how it renders directly in the [`react-live` documentation site](https://commerce.nearform.com/open-source/react-live):

<img width="1057" alt="image" src="https://github.com/facebook/docusaurus/assets/87080313/ebeaeab6-c250-4166-8f1d-1403ac2b6017">

### Actual behavior

live example doesn't render

<img width="500" alt="image" src="https://github.com/facebook/docusaurus/assets/87080313/d43a50d9-4b22-4544-8d37-177b77be58ff">

This can be also be reproduced directly in [docusaurus live docs](https://docusaurus.io/docs/markdown-features/code-blocks#interactive-code-editor)

Just paste the following "code" in the live editor

```jsx
function Example() {
  const something = {};

  return (
    <h3 style={{
      background: 'darkslateblue',
      color: 'white',
      padding: 8,
      borderRadius: 4
    }}>
      {`Hello World! 👋 ${something.foo?.bar ?? ''}`}
    </h3>
  )
}
```

### Your environment

- Public source code: n/a
- Public site URL: n/a
- Docusaurus version used: latest
- Browser: all
- Operating system and version: irelevant

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by csantos-nydig** [comment]
After more debugging, it seems the issue is directly in `react-live@v4.x`; I've opened an issue there as well: https://github.com/FormidableLabs/react-live/issues/389

(I think we should keep this issue open because docusaurus should upgrade react-live whenever their issue is resolved)

**Comment by Josh-Cena** [comment]
Duplicate of https://github.com/facebook/docusaurus/issues/9809. What's your react-live version? From that last comment it seems it's fixed in 4.1.6?

In any case, this will be released in a non-breaking version of react-live (or so I hope), so you can upgrade react-live without us bumping the package version.
